dask/tests/test_delayed.py

```diff
@@ -855,3 +856,27 @@ def add(x, y):
     assert add.__name__ == "add"
     assert add.__doc__ == "This is a docstring"
     assert add.__wrapped__(1, 2) == 3
+
+
+def test_delayed_fusion():
+    @delayed
+    def test(i):
+        return i + 1
+
+    @delayed
+    def test2(i):
+        return i + 2
+
+    @delayed
+    def test3(i):
+        return i + 3
+
+    obj = test3(test2(test(10)))
+    dsk = dict(collections_to_dsk([obj]))
+    assert len(dsk) == 3
+
+    obj2 = test3(test2(test(10)))
+    with dask.config.set({"optimization.fuse.delayed": True}):
+        dsk2 = dict(collections_to_dsk([obj]))
+    assert len(dsk2) == 2
+    assert dask.compute(obj) == dask.compute(obj2)
```

Comment:
Curious what you are testing here? (both `obj` and `obj2`) are defined as `test3(test2(test(10)))`.

Reply:
good point, I intended to compute the first on in the with statement, not sure why I didn't do that

Reply:
done